### PR TITLE
Progress towards #1237: Migrate front-end from GatsbyJS to Next.js 

### PR DESCRIPTION
**Pull request by manekenpix**
## Issue This PR Addresses
Progress towards #1237 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
We're switching to [Nextjs](https://nextjs.org/). Since keeping both frameworks in the same folder would be difficult for clean development and testing, this PR removes all the previous code and sets up the developer tools (linters, prettier, husky...) for `Nextjs`.

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Review by humphd — COMMENTED** [review]
I have a few questions, but this looks good.  I'm going to file a follow-up right now about what I think needs to be done on top of this, once this lands.

**Comment by humphd** [comment]
I've filed #1316 to deal with follow-ups from this PR.  Please let me know if I'm missing anything.

**Comment by humphd** [comment]
@manekenpix I see you pushing more, let me know when I should review again.

**Comment by manekenpix** [comment]
@humphd it's ready

File: src/frontend/.prettierrc
Comment (humphd) [reply src/frontend/.prettierrc:1]: Question: won't we run prettier from the root?  Why define this again if it's the same?

File: src/frontend/package.json
Comment (humphd) [inline src/frontend/package.json:9]: Similar question here: how does this play with our top-level eslint script?  Should we hook this up to the top-level eslint script as well?  Do we need to ignore anything in order to let this happen in the frontend differently?  Can eslint deal with different configs at different levels int he same run?

**Review by humphd — APPROVED** [review]
I have a few questions about how pretter/eslint at the top level will play with these changes, but we can experiment with this and adjust in follow-ups if you don't know the answer.

Great work on this, and thank you for helping us get this ready to go.  I'm excited for what's next! :)

File: src/frontend/package.json
Comment (manekenpix) [reply src/frontend/package.json:9]: To be honest, I'm not sure how it's going to work. My first idea was to run all the scripts from `src/frontend` and use `dev` or `prod` as backend since we won't be able to keep this branch synced with master (rebasing would be a nightmare), that's why I set up everything like if it were a separate repo. I think a `npm run backend-script && cd src/frontend && npm run frontend-script` approach might the easiest/safest one (if we don't completely ignore the backend), considering how different the backend and the frontend are from each other.

Comment (humphd) [reply src/frontend/package.json:9]: Let's just go for it and fix in follow-ups.

I'm happy for this to get merged if no one else wants to review further.

**Review by c3ho — APPROVED** [review]

**Pull request merged**
